src/compiler/transformers/ts.ts

```diff
@@ -1197,7 +1198,14 @@ export function transformTypeScript(context: TransformationContext) {
         // The names are used more than once when:
         //   - the property is non-static and its initializer is moved to the constructor (when there are parameter property assignments).
         //   - the property has a decorator.
-        if (isComputedPropertyName(name) && ((!hasStaticModifier(member) && currentClassHasParameterProperties) || hasDecorators(member) && legacyDecorators)) {
+        //   - the property is a method that has a parameter decorator
+        if (isComputedPropertyName(name) && (
+            !hasStaticModifier(member) && currentClassHasParameterProperties ||
+            legacyDecorators && (
+                hasDecorators(member) ||

```

Comment:
I've also experimented with using `nodeOrChildIsDecorated` locally, as I mentioned above, and the new test output for `decoratorsOnComputedProperties.ts` looks to be correct.

As I mentioned in https://github.com/microsoft/TypeScript/pull/53772#issuecomment-2276309207, could you change this to use `nodeOrChildIsDecorated` and pass through `parent` from each of the callers to this function?

This PR also needs an update from `main` as the `currentClassHasParameterProperties` check was removed in another PR.